modules/core/src/main/scala/gem/parser/Angle.scala

```diff
@@ -16,54 +17,48 @@ trait AngleParsers {
    * and at most 6 digits following the decimal point, and terminal parsers for each segment.
    */
   def genAngle(t1: Parser[_], t2: Parser[_], t3: Parser[_]): Parser[(Int, Int, Int, Int, Int)] =
-    for {
-      h  <- int <~ t1
-      m  <- int <~ t2
-      s  <- int <~ char('.')
-      µs <- frac(6)
-      _  <- t3
-    } yield (h, m, s, µs / 1000, µs % 1000)
+    (int <~ t1, int <~ t2, int <~ char('.'), frac(6) <~ t3) mapN { (h, m, s, µs) =>
+      (h, m, s, µs / 1000, µs % 1000)
+    } named s"genAngle($t2, $t2, $t3)"
```

Comment:
It may not be technically necessary, but the `for` comprehension syntax is easier to read, I think.  This is fine too though.